### PR TITLE
Assignment #3 - 100% coverage

### LINKED ISSUE #79: 100% coverage
Things are getting a bit more interesting...

It's time to add tests, verify coverage, confirm style, and automate all of this with CI. We will be using the [lab](https://github.com/hapijs/lab) module to perform all these tasks and automate it with [travis](https://travis-ci.org).

1. Add a `.travis.yml` file, testing our project on node 0.10, 0.12, and io.js (latest).
2. Add a test folder with two files, `version.js` and `index.js`, each testing the corresponding file under `/lib`.
3. Modify the `package.json` file to include the tests, as well as the dev dependency to lab.
4. Add the standard hapi.js Makefile to make it easy to generate other types of reports (e.g. HTML coverage report).
5. When using lab, enable coverage, require 100% coverage, enable linting with default rules, and use the [code](https://github.com/hapijs/code) assertion library.
6. Write a basic test to verify our version endpoint in `version.js`.
7. Change the `init()` method to accept a port and a callback. Use the callback to return when the function completes or errors. The `init()` callback should return any error state as well as a reference to the newly created server. This will allow us to later stop the server when we test it.
8. Export `init()` and move the invocation to a new `start.js` file (which will call the `init()` function with the `8000` port and a callback the outputs the information to the console when started). Change the `package.json` file to use the `start.js` file as the starting place. This file will not be covered by tests.
9. Write a basic test to verify the `init()` function in `index.js`.
10. Bring coverage to 100% by adding more tests as needed.

Everything up to (10) should be pretty straight forward. If you are not sure on how to use lab and code, look at other hapi.js modules like hoek, qs, items, and boom (e.g. simple modules) to copy their test scripts and setup.

Getting 100% coverage can be tricky sometimes so if you are not sure, get as much coverage as you can, and comment on the lines in your pull request where you are having a hard time reaching and someone will give you a clue.

Remember to properly `stop()` your servers when calling the `init()` method in each test.

For now, avoid using any of the `before()` and `after()` lab features.

As always, ask for help and help others!

Due: 4/4

### PR DESCRIPTION
**Pull request by hussainanjar**
Thanks to all whose PR are helping me do the assignments :+1: 

closes #79

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -5,24 +5,19 @@ var Hoek = require('hoek');
 var Version = require('./version');
 
 
-// Declare internals
+exports.init = function(port, callback) {
 
-var internals = {};
+	var server = new Hapi.Server();
```
Comment (hueniverse) [inline lib/index.js]: Use 4 spaces, not 8.

**Comment by zoe-1** [comment]
@hussainanjar Your Makefile is broken so tests are failing. 
In the Makefile commands must be tabbed in and not spaced indentation. 
Pretty simple to fix.

**Comment by hussainanjar** [comment]
@zoe-1 thanks for the detailed review will work on fixing these :+1:

**Comment by zoe-1** [comment]
@hussainanjar  good work on the changes. :-)

However, if I review your lib/index.js in assignment3 branch at:
https://github.com/hussainanjar/hueniversity/tree/assignment3
You still have Hoek variable assigned to required('hoek') but it is not used in the script.

**Comment by hussainanjar** [comment]
@zoe-1 for some reason I just couldn't see it, thanks for pointing out :+1:

**Pull request closed without merging**